### ISSUE TITLE
Allow `null` as default value for nullable strings

### ISSUE THREAD

**Issue by wedi** [stoplightio/spectral#1980]
**Describe the bug**

```
myProperty:
  type: string
  nullable: true
  default: null
```

fails validation with an obscure error message:

```
Cannot read properties of null (reading 'in')
error Command failed with exit code 2.
```

Setting `null` as default value is perfectly fine which has been clarified in the spec version 3.0.3.
See: https://github.com/OAI/OpenAPI-Specification/issues/2057#issuecomment-554777793 and https://github.com/OAI/OpenAPI-Specification/pull/2115#issuecomment-578115411.

**To Reproduce**

1. Given this OpenAPI/AsyncAPI document '...'
```shell
cat <<EOF > test.yaml
openapi: 3.0.3
components:
  schemas:
    myProperty:
      type: string
      default: null
      nullable: true
EOF
```

2. Run this CLI command '....'

`yarn spectral lint test.yaml`

3. See error

```shell
yarn spectral lint test.yaml
yarn run v1.22.17
$ /Users/weise/code/spectral_example/node_modules/.bin/spectral lint test.yaml
Cannot read properties of null (reading 'in')
error Command failed with exit code 2.
```

**Expected behavior**

- I expected the schema to pass validation.
- I expected a clearer error message but I'll fill another issue for this – probably buggy – behaviour.

**Environment (remove any that are not applicable):**
```shell
yarn spectral --version     
yarn run v1.22.17
$ /Users/weise/code/spectral_example/node_modules/.bin/spectral --version
6.1.0
✨  Done in 0.87s.
```

**Comment by P0lip** [comment]
This should be fixed by https://github.com/stoplightio/spectral/pull/1986.
LMK if it works.
Just make sure to upgrade `@stoplight/spectral-rulesets` beforehand.

**Comment by P0lip** [comment]
I'll close this one out in the meantime.
If you experience issues, please reach out here.
Thank you!

**Comment by wedi** [comment]
I can confirm it's solved Thanks! ❤️
